### ISSUE TITLE
Support for Phoenix Contact EVCC

### ISSUE THREAD

**Issue by Maverick78de** [snaptec/openWB#555]
Please add support for Phoenix Contact EVCC Controller with modbus TCP and RS485
This Controller ist used e.g. in wallb-e Wallboxes.

**Comment by Maverick78de** [comment]
If you need any documentation, I can provide it.

**Comment by snaptec** [comment]
wallb-e doesn't want to be supported, I'm sorry.

**Comment by Maverick78de** [comment]
I use the Phoenix Contact Controller which is open for buy, not only wallb-e use it! So it doesn't matter if wallb-e doesn't want to be supported.
e.g. https://esl-emobility.com/de/phoenix-contact-em-cp-pp-eth-2902802-zubehor-elektroinstallation-ladecontroller-ladesteuerung-wallbox.html

**Comment by Maverick78de** [comment]
Modbus documentation for the most common controllers can be found here:
https://www.phoenixcontact.com/online/portal/de/?uri=pxc-oc-itemdetail:pid=1018701&library=dede&pcck=P-29-04-02-01&tab=1&selectedCategory=ALL
and here 
https://www.phoenixcontact.com/online/portal/de/?uri=pxc-oc-itemdetail:pid=2902802&library=dede&pcck=P-29-04-02-01&tab=1&selectedCategory=ALL

I can give you a summary if you can tell me which things are required for communication with openwb

**Comment by snaptec** [comment]
I already contacted PhoenixContact if they would like to be integrated and will post the result here.

**Comment by snaptec** [comment]
Except from auto respond mails we get no answer.
The problem:
- We implement EVCC
- A customer has a problem
- We have no support from Phoenix Contact
- Customer says "openWB has a crappy support"

We are absolutely fine with implementing that if we have a designated contact at PC to escalate problems in short time.

**Comment by Maverick78de** [comment]
Ok .. Is it possible to implement a free configurable wallbox with http instead? So it could be possible to use any possible wallbox e.g. through ioBroker Rest API? You can mark this as unsupported method, so it could open openwb for more vendors.

**Comment by snaptec** [comment]
Yes, that is an option to enable advanced users a simple possibility to use different hardware. 
HTTP (get) Requests or MQTT?

**Comment by Maverick78de** [comment]
I personally would prefer HTTP GET which is easier to debug.

**Comment by snaptec** [comment]
No problem. I will add that feature and also the possibility to ask „plugged“ status through http „Ladeleistungsmodul“ as this is essential for proper function

**Comment by snaptec** [comment]
Please give it a try in 1.8.0.12 Nightly

**Comment by Maverick78de** [comment]
It looks good for the first try, but could it be possible to define the whole url for the setting like in Ladeleistungmodul -> "HTTP" ?

![Unbenannt](https://user-images.githubusercontent.com/19900958/86260014-05476700-bbbd-11ea-943c-70042235680c.PNG)

**Comment by andig** [comment]
@Maverick78de Phoenix it's implemented in evcc (although it has a known but at this time)

**Comment by mi-hol** [comment]
> is implemented in evcc

@snaptec Does this mean this issue can be closed?
